libs/live-network/src/network.ts

```diff
@@ -1,45 +1,43 @@
 import axios, {
-  AxiosPromise,
-  AxiosResponse,
+  type AxiosPromise,
+  type AxiosResponse,
+  type InternalAxiosRequestConfig,
   type AxiosError,
   type AxiosRequestConfig,
-  type Method,
 } from "axios";
 import { LedgerAPI4xx, LedgerAPI5xx, NetworkDown } from "@ledgerhq/errors";
 import { changes, getEnv } from "@ledgerhq/live-env";
 import { retry } from "@ledgerhq/live-promise";
 import { log } from "@ledgerhq/logs";
 
 type Metadata = { startTime: number };
-type ExtendedXHRConfig = AxiosRequestConfig & { metadata?: Metadata };
+type ExtendedXHRConfig = InternalAxiosRequestConfig & { metadata?: Metadata };
 
-export const requestInterceptor = (request: AxiosRequestConfig): ExtendedXHRConfig => {
+export const requestInterceptor = (request: ExtendedXHRConfig): ExtendedXHRConfig => {
   if (!getEnv("ENABLE_NETWORK_LOGS")) {
     return request;
   }
 
   const { baseURL, url, method = "", data } = request;
   log("network", `${method} ${baseURL || ""}${url}`, { data });
 
-  const req: ExtendedXHRConfig = request;
-
-  req.metadata = {
+  request.metadata = {
     startTime: Date.now(),
   };
 
-  return req;
+  return request;
 };
 
 type InterceptedResponse = {
   config: ExtendedXHRConfig;
-} & AxiosResponse<any>;
+} & AxiosResponse;
 
-export const responseInterceptor = (response: InterceptedResponse): InterceptedResponse => {
+export const responseInterceptor = (response: AxiosResponse): AxiosResponse => {
```

Comment:
Don't we want to override the `AxiosResponse` type as you did with `ExtendedXHRConfig` to avoid using type assertion, with something like:
`type ExtendedAxiosResponse = AxiosResponse & { config: { metadata?: Metadata } };`

Reply:
Sure ! Done 👍